src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -145,6 +146,40 @@ private void DownloadPackage(object sender, RoutedEventArgs e)
             packageInstaller.DownloadAndInstallPackage(package);
             DependencyRegen(currentWorkspace);
         }
+
+        /// <summary>
+        /// Handler of button which user click when choosing to keep the
+        /// installed version of package instead of the specified one.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void KeepLocalPackage(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                var info = ((PackageDependencyRow)((Button)sender).DataContext).DependencyInfo;
+                UpdateWorkspaceToUseInstalledPackage(info);
+            }
+            catch(Exception) {}
+        }
+
+        /// <summary>
+        /// Update current workspace in memory to keep the installed package
+        /// instead of keep referencing the dependency info saved in DYN
+        /// </summary>
+        /// <param name="info">Target PackageDependencyInfo to update version</param>
+        internal void UpdateWorkspaceToUseInstalledPackage(PackageDependencyInfo info)
+        {
+            var pmExtension = dependencyViewExtension.ViewLoadedParams.ExtensionManager.Extensions.Where(x => x.Name == "DynamoPackageManager").FirstOrDefault() as PackageManagerExtension;
+            if (pmExtension != null)
+            {
+                info.Version = new Version(pmExtension.PackageLoader.LocalPackages.Where(x => x.Name == info.Name).First().VersionName);
+                info.State = PackageDependencyState.Loaded;
+                // Mark the current workspace dirty for save
+                currentWorkspace.HasUnsavedChanges = true;
+                DependencyRegen(currentWorkspace);
```

Comment:
@QilongTang Is the expected behavior here for the `Keep Local` change to be serialized to the dyn as well? If so, I think you will need to remove (or replace) references to the old version of the package in the current workspace's  [nodePackageDictionary](https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs#L617).

Reply:
@scottmitchell Since we are updating the reference here directly, the package info in dictionary is updated at the same time, from my debugging we are good

Reply:
Oh good point!